src/test/rustdoc-ui/intra-doc/disambiguator-mismatch.stderr

```diff
@@ -2,94 +2,127 @@ error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:14:14
    |
 LL | /// Link to [struct@S]
-   |              ^^^^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^^^^ this link resolved to an enum, which is not a struct
    |
 note: the lint level is defined here
   --> $DIR/disambiguator-mismatch.rs:1:9
    |
 LL | #![deny(rustdoc::broken_intra_doc_links)]
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   = note: this link resolved to an enum, which is not a struct
+help: to link to the enum, prefix with `enum@`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:19:14
    |
 LL | /// Link to [mod@S]
-   |              ^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^ this link resolved to an enum, which is not a module
+   |
+help: to link to the enum, prefix with `enum@`
    |
-   = note: this link resolved to an enum, which is not a module
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:24:14
    |
 LL | /// Link to [union@S]
-   |              ^^^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^^^ this link resolved to an enum, which is not a union
+   |
+help: to link to the enum, prefix with `enum@`
    |
-   = note: this link resolved to an enum, which is not a union
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:29:14
    |
 LL | /// Link to [trait@S]
-   |              ^^^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^^^ this link resolved to an enum, which is not a trait
    |
-   = note: this link resolved to an enum, which is not a trait
+help: to link to the enum, prefix with `enum@`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `T`
   --> $DIR/disambiguator-mismatch.rs:34:14
    |
 LL | /// Link to [struct@T]
-   |              ^^^^^^^^ help: to link to the trait, prefix with `trait@`: `trait@T`
+   |              ^^^^^^^^ this link resolved to a trait, which is not a struct
+   |
+help: to link to the trait, prefix with `trait@`
    |
-   = note: this link resolved to a trait, which is not a struct
+LL | /// Link to [trait@T]
+   |              ^^^^^^
 
 error: incompatible link kind for `m`
   --> $DIR/disambiguator-mismatch.rs:39:14
    |
 LL | /// Link to [derive@m]
-   |              ^^^^^^^^ help: to link to the macro, add an exclamation mark: `m!`
+   |              ^^^^^^^^ this link resolved to a macro, which is not a derive macro
+   |
+help: to link to the macro, add an exclamation mark
    |
-   = note: this link resolved to a macro, which is not a derive macro
+LL | /// Link to [derive@m!]
+   |                      ^
```

Comment:
We should be removing the `derive@`, right?

Reply:
Yes! I'm working on it. I simply pushed the another change you suggested (about the `note`).